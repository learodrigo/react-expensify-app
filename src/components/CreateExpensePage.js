import React from "react";

const CreateExpensePage = () => (
  <div>
    This is from my Add
  </div>
);

export default CreateExpensePage;