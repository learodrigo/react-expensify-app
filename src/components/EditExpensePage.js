import React from "react";

const EditExpensePage = () => (
  <div>
    This is from my Edit
  </div>
);

export default EditExpensePage;