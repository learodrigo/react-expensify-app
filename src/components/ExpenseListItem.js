import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>Qty: <strong>{amount}</strong> - Date: <strong>{createdAt}</strong></p>
    
  </div>
);

export default ExpenseListItem;