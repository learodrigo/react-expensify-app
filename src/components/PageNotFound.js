import React from "react";
import { Link } from 'react-router-dom';

const PageNotFound = () => (
  <div>
    <h3>Error 404 - Page not found :(</h3>
    <Link to="/">Dashboard</Link>
  </div>
);

export default PageNotFound;