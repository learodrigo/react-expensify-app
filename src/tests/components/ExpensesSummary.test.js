import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render ExpensesSummary with 1 expense correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with multiple expenses correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={123456789321} />);
  expect(wrapper).toMatchSnapshot();
});