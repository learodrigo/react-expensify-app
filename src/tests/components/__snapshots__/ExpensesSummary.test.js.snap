// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     - Total 
    $2.35
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with multiple expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     - Total 
    $1,234,567,893.21
  </h1>
</div>
`;
